doc/flame/collision_detection.md

```diff
@@ -1,137 +1,143 @@
-hitboxes and to more accurately recognize gestures on top of your `PositionComponent`s. Say that you
-have a fairly round rock as a `SpriteComponent` for example, then you don't want to register input
-that is in the corner of the image where the rock is not displayed, since an image is always
-rectangular. Then you can use the `HasHitboxes` mixin to define a more accurate polygon (or another
-shape) for which the input should be within for the event to be registered on your component.
-
-You can add new shapes to the `HasHitboxes` just like they are added in the below `Collidable`
-example.
-
-### Collidable
-The `Collidable` mixin is added to a `PositionComponent` that has a `HasHitboxes` and it is used for
-detecting collisions with other `Collidable`s. If you do not add a shape to your `HasHitboxes` 
-component it will never collide with anything. If you want the component to have a default 
-rectangular shape that fills the size of your component you can simply do 
-`addHitbox(HitboxRectangle())`.
-
-To make your component collidable you would start off something like this:
+
+### HasCollisionDetection
+
+If you want to use collision detection in your game you have to add the `HasCollisionDetection`
+mixin to your game so that the game knows that it should use a collision detection system to keep
+track of which components that can collide.
+
+Example:
 
 ```dart
-class MyCollidable extends PositionComponent with HasHitboxes, Collidable {
-  MyCollidable() {
-    // This could also be done in onLoad instead of in the constructor
-    final shape = HitboxPolygon([
-      Vector2(0, 1),
-      Vector2(1, 0),
-      Vector2(0, -1),
-      Vector2(-1, 0),
-    ]);
-    addHitbox(shape);
-  }
+class MyGame extends FlameGame with HasCollisionDetection {
+  // ...
 }
 ```
 
-The `HitboxPolygon` added to the `Collidable` here is a diamond shape(◇).
-More about how the different shapes are defined in the [Shapes](#shapes) section.
+Now when you add `HitboxShape`s to components that are then added to the game, they will
+automatically be checked for collisions.
 
-Remember that you can add as many `HitboxShape`s as you want to your `Collidable` to make up more
-complex hitboxes. For example a snowman with a hat could be represented by three `HitboxCircle`s and
-a `HitboxPolygon` as its hat.
 
-To react to a collision you should override the `collisionCallback` in your component.
+### CollisionCallbacks
+
+To react to a collision you should add the `CollisionCallbacks` mixin to your component.
 Example:
 
 ```dart
-class MyCollidable extends PositionComponent with HasHitboxes, Collidable {
-  ...
-
+class MyCollidable extends PositionComponent with CollisionCallbacks {
   @override
-  void onCollision(Set<Vector2> points, Collidable other) {
-    if (other is CollidableScreen) {
-      ...
-    } else if (other is YourOtherCollidable) {
-      ...
+  void onCollision(Set<Vector2> points, PositionComponent other) {
```

Comment:
`onCollisionStart`?

Also: I wonder if we should wrap these into an "Event" object, similar to how we have events for taps, drags, key presses, and so on.